docs/guides/vs-code-debugging.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Debugging With VS Code
+sidebar_label: VS Code Debugging
+description: How to debug a Stencil component using VS Code
+slug: /vs-code-debugging
+---
+
+# Debugging With VS Code
+
+VS Code offers a streamlined debugging experience that can be started with a single click when using [launch configurations](https://code.visualstudio.com/docs/editor/debugging#_launch-configurations).
+
+For more information on how to use the VS Code debugger, please visit the [debugger documentation](https://code.visualstudio.com/docs/editor/debugging).
+
+## Requirements for Debugging
+
+In order for a debugger to function, the Stencil project must be configured to generate source maps for the compiled web components back to the source code. As of Stencil v3, source maps are generated by default, but be sure to double check the project's Stencil config does not disable this behavior. More information regarding source maps in Stencil can be found in the [Stencil documentation](https://stenciljs.com/docs/config#sourcemap).
+
+## Debugging Stencil Components In a Web App
+
+It's a common use case to want to step through a web component's code as it executes in the browser and VS Code makes that process simple. Combining the debugger with Stencil's dev server in watch mode will allow you to debug changes as they're made.
+
+### Configuring the VS Code Debugger
+
+To debug Stencil components as they run in a browser (web app), create (or edit) the `launch.json` file with the following configuration:
+
+```json title="launch.json"
```

Comment:
Similar to my point above, should we clarify this file is often found in `.vscode/` in a project's root?
```suggestion
```json title=".vscode/launch.json"
```